### ISSUE TITLE
`tls: bad certificate` error emitted on server every time agent connects

### ISSUE THREAD

**Issue by brandond** [k3s-io/k3s#4067]
Whenever a node connects to a server, it does so first using the OS CA bundle, before retrying with the cluster CA bundle. The original connection is expected to fail, as the cluster CA is not part of the OS CA bundle.This failure causes a log message that frequently causes users to think that there is something wrong with their cluster:

`Sep 22 23:32:04 ip-172-31-35-30 rke2[12958]: time="2021-09-22T23:32:04Z" level=info msg="Cluster-Http-Server 2021/09/22 23:32:04 http: TLS handshake error from 127.0.0.1:44528: remote error: tls: bad certificate"`

We should make this log message debug only, and maybe emit a more specific message when agents connect to tell the user what's going on.

**Comment by galal-hussein** [comment]
I dont think we can do that, since the TLS handshake happens before forwarding the request to the cacerts handler, then the listener will send this message, and there is no way to identify that early if this request is coming from a startup agent just a normal tls handshake error, of course we can still catch any handshake error by doing something like:

```
// Create a HTTP server with the registered request handlers, using logrus for logging
	server := http.Server{
		Handler:  handler,
		ErrorLog: log.New(&writerAdapter{logrus.StandardLogger().Writer()}, "Cluster-Http-Server ", log.LstdFlags),
	}
```

and then:

```
var tlsHandshakeError = "http: TLS handshake error"

func (a *writerAdapter) Write(data []byte) (int, error) {
	if strings.Contains(string(data), tlsHandshakeError) {
		logrus.Debugf("HANDSHAKE ERROR %v", string(data))
	}
	return a.Write(data)
}
```
but do we really want to turn off all TLS handshake errors and make them in debug log mode only? @brandond

**Comment by brandond** [comment]
I've literally never had this error help me with anything, and it causes a ton of confusion. If there is a legitimate problem with the client trusting the cluster CA certificate, that is reported client-side; also capturing it server-side isn't worth the added confusion.
